homeassistant/components/dyson/fan.py

```diff
@@ -237,28 +237,19 @@ def oscillate(self, oscillating: bool) -> None:
     @property
     def oscillating(self):
         """Return the oscillation state."""
-        return self._device.state and self._device.state.oscillation == "ON"
+        return self._device.state.oscillation == "ON"
 
     @property
     def is_on(self):
         """Return true if the entity is on."""
-        if self._device.state:
-            return self._device.state.fan_mode in ["FAN", "AUTO"]
-        return False
+        return self._device.state.fan_mode in ["FAN", "AUTO"]
 
     @property
     def speed(self) -> str:
         """Return the current speed."""
-        if self._device.state:
-            if self._device.state.speed == FanSpeed.FAN_SPEED_AUTO.value:
-                return self._device.state.speed
-            return int(self._device.state.speed)
-        return None
-
-    @property
-    def current_direction(self):
-        """Return direction of the fan [forward, reverse]."""
-        return None
+        if self._device.state.speed == FanSpeed.FAN_SPEED_AUTO.value:
+            return self._device.state.speed
```

Comment:
We should change this in a follow up to adhere to our standard fan speeds.

https://developers.home-assistant.io/docs/core/entity/fan#properties

Reply:
I'll handle that in a follow up PR.